test/nl/surf/eduhub_rio_mapper/e2e_helper.clj

```diff
@@ -0,0 +1,349 @@
+;; - application added to gateway to access test entities
+
+
+
+(def ^:private last-seen-testing-contexts (atom nil))
+
+(defn- print-testing-contexts
+  "Print eye catching testing context (when it's not already printed)."
+  []
+  (let [testing-contexts (test/testing-contexts-str)]
+    (when-not (= @last-seen-testing-contexts
+                 testing-contexts)
+      (reset! last-seen-testing-contexts testing-contexts)
+      (println "\n###\n###" testing-contexts "\n###\n"))))
+
+(defmacro print-boxed
+  "Print pretty box around output of evaluating `form`."
+  [title & form]
+  `(let [s# (with-out-str (do ~@form))]
+     (print-testing-contexts)
+     (print "╭─────" ~title "\n│ ")
+     (println (str/replace (str/trim s#) #"\n" "\n│ "))
+     (println "╰─────")))
+
+(defn- print-soap-body
+  "Print the body of a SOAP request or response."
+  [s]
+  (let [xml (xml/parse (ByteArrayInputStream. (.getBytes s)))]
+    (xml-utils/pretty-print-xml (-> xml :content second :content first)
+                                :initial-indent "  ")))
+
+(defn- print-json
+  "Print indented JSON."
+  [v]
+  (when v
+    (print "  ")
+    (println (json/write-str v :indent true :indent-depth 1))))
+
+(defn- print-json-str
+  "Parse string as JSON and print it."
+  [s]
+  (when s
+    (print-json (json/read-str s))))
+
+(defn- print-api-message
+  "Print boxed API request and response."
+  [{{:keys [method url]}  :req
+    {:keys [status body]} :res}]
+  (print-boxed
+      "API"
+    (println (str/upper-case (name method)) url status)
+    (when body
+      (print-json body))))
+
+(defn- print-rio-message
+  "Print boxed RIO request and response."
+  [{{:keys                [method url]
+     req-body             :body
+     {action :SOAPAction} :headers} :req
+    {res-body :body
+     :keys    [status]}             :res}]
+  (print-boxed
+      "RIO"
+    (println (str/upper-case method) url status)
+    (println "- action:" action)
+    (println "- request:\n")
+    (print-soap-body req-body)
+    (println)
+    (when (= http-status/ok status)
+      (println "- response:\n")
+      (print-soap-body res-body)
+      (println))))
+
+(defn- print-ooapi-message
+  "Print boxed OOAPI request and response."
+  [{{:keys [method url]}  :req
+    {:keys [status body]} :res}]
+  (print-boxed
+      "OOAPI"
+    (println (str/upper-case method) url status)
+    (println)
+    (when (= http-status/ok status)
+      (print-json-str body))))
+
+(defn- print-http-messages
+  "Print HTTP message as returned by API status."
+  [http-messages]
+  (when-let [{:keys [req] :as msg} (first http-messages)]
+    (if (and (= "post" (:method req))
+             (-> req :headers :SOAPAction))
+      (print-rio-message msg)
+      (print-ooapi-message msg))
+    (recur (next http-messages))))
+
+
+
+;; Defer running make-config so running some (other!) tests is still
+;; possible when environment incomplete.
+(def config (memoize #(cli/make-config)))
```

Comment:
I would use `delay` instead of `memoize` for this. Then you'd use it as `@config`